### DISCUSSION TITLE
[Scripting] How to split `"abc"` to `("a", "b", c")`

### DISCUSSION THREAD

**Discussion by goekce** [typst/typst#2888] [Q&A] [answered]
I want to
```typst
#"XOX".split("") 
```
and it should output `("X", "O", "X")`, but I get `("", "X", "O", "X", "")`. I found out that this is a feature of the [rust `split` method](https://doc.rust-lang.org/std/primitive.str.html#method.split). I could solve the problem using [the empty match `\B`](https://docs.rs/regex/latest/regex/#empty-matches):

```typst
#"XOX".split(regex("\B"))
```

Does someone have a more readable solution?

**Comment by frozolotl** [comment, marked as answer]
You should use `str.clusters()` instead, though `str.codepoints()` would also work in many cases.
`str.clusters()` is usually the correct choice. See [this blog post](https://manishearth.github.io/blog/2017/01/14/stop-ascribing-meaning-to-unicode-code-points) for more information on grapheme clusters.
